### ISSUE TITLE
How to refresh JWT token C# REST Client

### ISSUE THREAD

**Issue by pranavd** [docusign/docusign-csharp-client#171]
Hi, I am using JWT authentication from C# REST SDK. My question is regarding refreshing of JWT after expiry. Will I get the new token automatically from the API calls, or I have to refresh it separately. In case if I have to refresh it separately then I guess I have to place the lock on header until it is refreshed so that no other API call can use it

**Comment by LarryKlugerDS** [comment]
You (the client app) need to refresh it. Auto-refresh is on the roadmap but no date yet.

I suggest that you check the expiration date before each call, then proactively refresh (actually recreate) if you have less than 10 minutes left. This should be done before every call. 

If multiple threads do the check then you'd need a mutex on the check/update method (if you're using the SDK from multiple threads).

Also note that due to security-related issues at either the account or platform (DocuSign) level, a token may suddenly stop working at any time (eg before the expiration timestamp). So be ready to handle that. -- Same sort of issue as a network failure.

**Comment by RajRele** [comment]
@pranavd Closing this issue, if you still have questions, please comment below or create a new issue. Thanks.
